### PR TITLE
Add basic mapping for query

### PR DESCRIPTION
**Pull request by rcantin-w**
I could add more later on, such as the analyzers in settings and the `fields` sections in the mapping.

Could also add `body` and `standfirst` in the next PR.

You can run `GET /testing/_mapping` to see the current state.

Currently results in:

```
{
  "_index": "testing",
  "_id": "ZBhbrxQAAB3yH6W6",
  "_version": 1,
  "_seq_no": 0,
  "_primary_term": 1,
  "found": true,
  "_source": {
    "id": "ZBhbrxQAAB3yH6W6",
    "display": {
      "type": "Article",
      "id": "ZBhbrxQAAB3yH6W6",
      "title": "Translation",
      "caption": "It can be OK to ask a direct question… sometimes skirting around the topic of cancer can make it the elephant in the room that hangs around awkwardly.",
      "format": {
        "type": "ArticleFormat",
        "id": "W7d_ghAAALWY3Ujc",
        "label": "Comic"
      },
      "publicationDate": "2023-03-31T09:00:00+0000",
      "contributors": [
        {
          "type": "Contributor",
          "contributor": {
            "type": "Person",
            "id": "Y8aT1REAAKREABkW",
            "label": "Alex Brenchley"
          },
          "role": {
            "type": "EditorialContributorRole",
            "id": "Wux6DyIAAO5n3lzk",
            "label": "Artist"
          }
        }
      ],
      "image": {
        "type": "PrismicImage",
        "dimensions": {
          "width": 4500,
          "height": 4500
        },
        "alt": "Translation | | | | | Alex Brenchley for Wellcome Collection |",
        "copyright": "Panel 2 of a six-panel comic drawn digitally: Speech bubbles read \"How... [long pause] are you?*\" and \"How are... errm... things?*\". The word things is italicised. The asterisk below says \"Please don't make me use the word 'cancer'!\"",
        "url": "https://images.prismic.io/wellcomecollection/ac67f7fc-c6a2-42c5-8ed9-142c771cecfe_LAC+Panel+10.2.png?auto=compress,format",
        "32:15": {
          "dimensions": {
            "width": 3200,
            "height": 1500
          },
          "alt": "Translation | | | | | Alex Brenchley for Wellcome Collection |",
          "copyright": "Panel 2 of a six-panel comic drawn digitally: Speech bubbles read \"How... [long pause] are you?*\" and \"How are... errm... things?*\". The word things is italicised. The asterisk below says \"Please don't make me use the word 'cancer'!\"",
          "url": "https://images.prismic.io/wellcomecollection/ac67f7fc-c6a2-42c5-8ed9-142c771cecfe_LAC+Panel+10.2.png?auto=compress,format&rect=0,2178,4500,2109&w=3200&h=1500"
        },
        "16:9": {
          "dimensions": {
            "width": 3200,
            "height": 1800
          },
          "alt": "Translation | | | | | Alex Brenchley for Wellcome Collection |",
          "copyright": "Panel 2 of a six-panel comic drawn digitally: Speech bubbles read \"How... [long pause] are you?*\" and \"How are... errm... things?*\". The word things is italicised. The asterisk below says \"Please don't make me use the word 'cancer'!\"",
          "url": "https://images.prismic.io/wellcomecollection/ac67f7fc-c6a2-42c5-8ed9-142c771cecfe_LAC+Panel+10.2.png?auto=compress,format&rect=0,1969,4500,2531&w=3200&h=1800"
        },
        "square": {
          "dimensions": {
            "width": 3200,
            "height": 3200
          },
          "alt": "Translation | | | | | Alex Brenchley for Wellcome Collection |",
          "copyright": "Panel 2 of a six-panel comic drawn digitally: Speech bubbles read \"How... [long pause] are you?*\" and \"How are... errm... things?*\". The word things is italicised. The asterisk below says \"Please don't make me use the word 'cancer'!\"",
          "url": "https://images.prismic.io/wellcomecollection/ac67f7fc-c6a2-42c5-8ed9-142c771cecfe_LAC+Panel+10.2.png?auto=compress,format&rect=0,0,4500,4500&w=3200&h=3200"
        }
      }
    },
    "query": {
      "title": "Translation",
      "published": "2023-03-31T09:00:00.000Z",
      "contributors": [
        "Alex Brenchley"
      ],
      "promo_caption": "It can be OK to ask a direct question… sometimes skirting around the topic of cancer can make it the elephant in the room that hangs around awkwardly."
    }
  }
}
```

### CODE REVIEW

File: pipeline/src/helpers/elasticsearch.ts [hunk 1]
```diff
@@ -30,6 +60,9 @@ export const bulkIndexDocuments = async <T extends HasIdentifier>(
         index: { _index: indexName, _id: doc.id },
       };
     },
+    onDrop(fail) {
+      console.log(fail);
```
Comment (jamieparkinson) [inline pipeline/src/helpers/elasticsearch.ts:64]: Would be helpful if this log was more descriptive

File: pipeline/src/types/index.ts [hunk 2]
```diff
@@ -35,6 +35,17 @@ export type { PrismicImage };
...
 export type { Article, Contributor, ArticleFormat, Image };
 
+export type ElasticsearchArticle = {
+  id: string;
+  display: Article;
+  query: {
+    title: string;
+    published: Date;
+    contributors: string[];
+    promo_caption?: string;
```
Comment (jamieparkinson) [inline pipeline/src/types/index.ts:45]: I would expect this to have the same name that it does in the display model - is there a reason it's different?

File: pipeline/src/helpers/elasticsearch.ts [hunk 3]
```diff
@@ -3,12 +3,42 @@ import { Readable } from "stream";
...
+                type: "text",
+              },
+              promo_caption: {
+                type: "text",
+              },
+            },
+          },
+        },
+      },
+    });
```
Comment (jamieparkinson) [inline pipeline/src/helpers/elasticsearch.ts:41]: Can we put the mapping in its own separate file?

File: pipeline/src/types/index.ts (hunk 2 above)
Comment (rcantin-w) [reply pipeline/src/types/index.ts:45]: Happy to change that, actually wanted to and had already approved it with Harrison (these are the names in his prototype), so changing it now.

File: pipeline/src/helpers/prismic.ts [hunk 4]
```diff
@@ -45,7 +66,7 @@ export const getPrismicDocuments = async <T>({
   contentTypes,
   after,
 }: GetPrismicDocumentsParams): Promise<PrismicPage<T>> => {
-  const docs = await client.getByType(contentTypes[0], {
+  const docs = await client.get({
```
Comment (rcantin-w) [inline pipeline/src/helpers/prismic.ts:69]: @jamieparkinson fyi; just realised `webcomics` were still being skipped on. Now using simple `get` and using the predicates to control which content types to fetch.

**Pull request merged**